Documentation/kubernetes-networking.md

```diff
@@ -0,0 +1,91 @@
+| TCP      | 2379       | Control Nodes | etcd server client API                                   |
+| TCP      | 2379       | Worker Nodes  | etcd server client API (only required if using flannel). |
+| TCP      | 2380       | etcd Nodes    | Used for etcd intra-cluster communication.               |
+
+[external-service]: http://kubernetes.io/v1.0/docs/user-guide/services.html#external-services
+
+## Advanced Configuration
+
+The Coreos Kubernetes documentation will describe either using cloud-provider networking integration (e.g. AWS, GCE), or a software-defined overlay network (e.g. [flannel][coreos-flannel]) to manage the Kubernetes Pod Network. However, in some cases a deployer may want to make use of existing network infrastructure to manage the Kubernetes network themselves.
+
+[coreos-flannel]: https://coreos.com/flannel/docs/latest/flannel-config.html
+
+### Container-to-Container Communication
+
+No specific configuration is necessary, as the networking between containers in a pod is managed by Kubernetes itself. This is accomplished by linking all containers in a pod into the same local network.
+
+### Pod-to-Pod Communication
+
+Each pod in the Kubernetes cluster will be assigned an IP that is expected to be routable from all other hosts and pods in the Kubernetes cluster. Generally this will be accomplished by assigning an IP range to each host, and allocating an IP from that range to each launched pod.
+
+Example:
+
+* Node A assigned 10.0.1.0/24
+* Node B assigned 10.0.2.0/24.
+
+When a Pod is launched on `Node A` it might be assigned `10.0.1.33` and on `Node B` a pod could be assigned `10.0.2.144`.
+It would then be expected that both pods would be able to reach each other via those IPs, as if they were on a flat network.
+
+To achieve this network model, there are various methods that can be used. See the [Kubernetes Networking][how-to-achieve] documentation for more detail.
+
+[how-to-achieve]: http://kubernetes.io/v1.0/docs/admin/networking.html#how-to-achieve-this
+
+### Pod-to-Service Communication
+
+The service IPs are assigned from a range configured in the Kubernetes api-server via the `--service-ip-range` flag. These are virtual IPs which are intercepted by a kube-proxy process running locally on each node. These IPs do not need to be routable off-host, because IPTables rules will intercept the traffic, and route to the proper backend (usually the pod network).
+
+A requirement of a manually configured network is that the service-ip-range does not conflict with existing network infrastructure.
+
+### External-to-Internal Communication
+
+IPs assigned on the pod network are usually not going to be routable outside of the cluster. Generally, allowing external traffic into the cluster will be managed by mapping external load-balancers to route requests to all nodes in the cluster, allowing the kube-proxy process to then route requests to the proper pods using the cluster pod-network.
```

Comment:
The "all nodes" part of this reads weirdly. I'm thinking its because its just an implementation detail...you could select a subset of the machines.

I feel like its important to mention that only _specific_ services are expose. Before it read that tons of stuff was exposed, between the "all" in "all nodes" and the mention of a wide port range above.

What about:

"IPs assigned on the pod network are usually not going to be routable outside of the cluster. **This isn't an issue since most communication between your applications stays within the cluster, as described above.**

Allowing external traffic into the cluster **is generally accomplished** by mapping external load-balancers to **specifically exposed services** in the cluster. This mapping allows the kube-proxy process to route the **external requests** to the proper pods using the **cluster's** pod-network."